libs/comparator/tables.js

```diff
@@ -39,6 +39,13 @@ const buildColumnsMap = table => {
                 column.unique && (options.push('UQ'))
                 column.required && (options.push('NN'))
 
+                if (relation.metaInfo?.relationIdentificationColumnId) {
```

Comment:
There should be no **columnId** on this level of the logic.

Once you retrieved the tables/columns information from backendless - immediately replace columnId by columnName.
This will give you the possibility
- to dump schema without any additional logic
- to compare live apps without any additional logic
- to compare json schemas without any additional logic
- all logic will sit in the only place - backendless-console-api

the column id to name transformation should happen in the getAppDataTables->normalizeTable function

the column name to column id transformation should happen in the 
updateColumn method

Reply:
@vengrov  thank you, I did not come to this myself... please check the refactored version. I tested it locally, seems to be working fine